lib/watir-webdriver/element_collection.rb

```diff
@@ -89,16 +89,33 @@ def to_a
     def elements
       @parent.is_a?(IFrame) ? @parent.switch_to! : @parent.send(:assert_exists)
 
-      @elements ||= locator_class.new(
-        @parent.wd,
-        @selector,
-        element_class.attribute_list
-      ).locate_all
+      element_validator = element_validator_class.new
+      selector_builder = selector_builder_class.new(@parent, @selector, element_class.attribute_list)
+      locator = locator_class.new(@parent, @selector, selector_builder, element_validator)
+
+      @elements ||= locator.locate_all
     end
 
-    # overridable by subclasses
     def locator_class
-      ElementLocator
+      Kernel.const_get("#{element_class}::Locator")
```

Comment:
We should use #const_get on the immediate parent of the element class rather than using Kernel